### PR TITLE
feat(json-body-parser): allow specifying versioned APIGW events

### LINKED ISSUE #1085: TypeScript improvement: `@middy/http-json-body-parser` should support API Gateway V2 events
As discussed in #1023 and suggested by @apalchys:

---

List of issues my team see (I'm going to keep it up to date):

- `@middy/http-json-body-parser` defines event using legacy `APIGatewayEvent` which describes event payload format v1. No Payload format v2 support.

_Originally posted by @apalchys in https://github.com/middyjs/middy/issues/1023#issuecomment-1484228944_

### PR DESCRIPTION
**Pull request by naorpeled**
closes #1085

### CODE REVIEW

File: packages/http-json-body-parser/index.d.ts [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 
 interface Options {
   reviver?: (key: string, value: any) => any
   disableContentTypeError?: boolean
 }
 
-export type Event = Omit<APIGatewayEvent, 'body'> & {
+type VersionedApiGatewayEvent = APIGatewayEvent | APIGatewayProxyEventV2
+
+export type Event<APIGatewayEventType extends VersionedApiGatewayEvent = VersionedApiGatewayEvent> = Omit<APIGatewayEventType, 'body'> & {
```
Comment (naorpeled) [inline packages/http-json-body-parser/index.d.ts:12]: BTW, I think that at some point we should rename this to ParsedEvent,
not super crucial, but it conveys its meaning more precisely.

@lbertenasco wdyt?

Comment (lmammino) [reply packages/http-json-body-parser/index.d.ts:12]: I like the suggestion, can we start by doing a type alis and deprecating the old one? Thoughts @willfarrell ?

**Review by lmammino — APPROVED** [review]
This looks good to me! Thanks a million for taking the time @naorpeled !

@willfarrell , should we target this to the `main` branch or is this best to go together with `v5`?

I guess the choice probably depends on whether we'd like to rename `Event` to `ParsedEvent`...

**Comment by willfarrell** [comment]
> should we target this to the main branch or is this best to go together with v5?

Let's put it all in v5, it will be soon enough.

**Pull request merged**